private/bufpkg/bufcheck/bufbreaking/bufbreaking.go

```diff
@@ -91,6 +91,30 @@ func GetAllRulesV1() ([]bufcheck.Rule, error) {
 	return rulesForInternalRules(internalConfig.Rules), nil
 }
 
+// GetAllRulesAndCategoriesV1 returns all rules and categories for v1 as a string slice.
+//
+// This is used for validation purposes only.
+func GetAllRulesAndCategoriesV1() []string {
+	var allRulesAndCategories []string
+	allRulesAndCategories = append(allRulesAndCategories, bufbreakingv1.VersionSpec.AllCategories...)
+	for id := range bufbreakingv1.VersionSpec.IDToCategories {
```

Comment:
Is it going to be an issue that the order is non-deterministic?

Reply:
For now, no, since it ends up being converted into a map for look-ups in the validation logic... I wonder if it's worth making it deterministic here anyway, just as a matter of consistency?

Reply:
I agree there is no reason to sort if it just goes into a map.